httpx/_urlparse.py

```diff
@@ -449,6 +446,39 @@ def quote(string: str, safe: str = "/") -> str:
     )
 
 
+def quote(string: str, safe: str = "/") -> str:
```

Comment:
I believe we should also add unit tests for these functions, rather than simply testing them with `httpx.URL`. 
This would be a more robust approach, in my opinion.

Reply:
IMO current approach is fine.

> This is clearly a code-smell, because our test cases ought to be tests against our public API, rather than testing implementation details. Perhaps there's some cases where it's a necessary hack, but... perhaps not?


from https://github.com/encode/httpx/issues/2492#issue-1478857204

Reply:
I agree testing the public API should be sufficient unless something private is particularly expensive to test via the public API.

Reply:
It's a matter of preference, but if we encounter regression in our `httpx.URL` tests, we will go through these functions and find the method that isn't working properly, which is why unit tests are useful.